src/Microsoft.TestPlatform.PlatformAbstractions/common/Tracing/PlatformEqtTrace.cs

```diff
@@ -147,6 +175,11 @@ public bool InitializeVerboseTrace(string customLogFile)
         /// <inheritdoc/>
         public bool ShouldTrace(PlatformTraceLevel traceLevel)
         {
+            if (DoNotInitialize)
+            {
+                return false;
+            }
+
             switch (traceLevel)
             {
                 case PlatformTraceLevel.Off:
```

Comment:
<Spoke to harsh offline>
We plan to cache the current PlatformTraceSource value and use that to bail out faster checking for `PlatformTraceSource.Off`.